### PR TITLE
TINY-10058: Add align property to labels to control text alignment

### PR DESCRIPTION
**Pull request by mkzhx**
Related Ticket: TINY-10058

Description of Changes:
* Add align property to labels to control text alignment

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/themes/silver/main/ts/ui/dialog/Label.ts [hunk 1]
```diff
@@ -9,10 +9,11 @@ import { RepresentingConfigs } from '../alien/RepresentingConfigs';
 type LabelSpec = Omit<Dialog.Label, 'type'>;
 
 export const renderLabel = (spec: LabelSpec, backstageShared: UiFactoryBackstageShared): SimpleSpec => {
+  const endClass = spec.align === 'end' ? [ 'tox-label--end' ] : [];
   const label: AlloySpec = {
     dom: {
       tag: 'label',
-      classes: [ 'tox-label' ]
+      classes: [ 'tox-label', ...endClass ]
```
Comment (MitchCTiny) [inline modules/tinymce/src/themes/silver/main/ts/ui/dialog/Label.ts:16]:
```suggestion
  const classes = [ 'tox-label',  ...(spec.align === 'end' ? [ 'tox-label--end' ] : []) ];
  const label: AlloySpec = {
    dom: {
      tag: 'label',
      classes
```

**Review by MitchCTiny — APPROVED** [review]
Smashing these PRs out!

File: modules/bridge/src/main/ts/ephox/bridge/components/dialog/Label.ts [hunk 2]
```diff
@@ -7,16 +7,19 @@ export interface LabelSpec {
   type: 'label';
   label: string;
   items: BodyComponentSpec[];
+  align?: 'start' | 'end';
```
Comment (TheSpyder) [inline modules/bridge/src/main/ts/ephox/bridge/components/dialog/Label.ts:10]: I'm happy to add `center`, if you think it's valuable

**Review by TheSpyder — APPROVED** [review]

File: modules/bridge/src/main/ts/ephox/bridge/components/dialog/Label.ts (hunk 2 above)
Comment (MitchCTiny) [reply modules/bridge/src/main/ts/ephox/bridge/components/dialog/Label.ts:10]: Center is the default when the label is in a grid component, as in the ai dialog. I forgot about that, but we should probably implement a forced alignment for start as well, and center could be added for completeness.

Comment (mkzhx) [reply modules/bridge/src/main/ts/ephox/bridge/components/dialog/Label.ts:10]: The label is still has start text alignment in a grid component, so there is no concern making `align: 'start'` the default. I have also added center alignment.

File: modules/tinymce/src/themes/silver/main/ts/ui/dialog/Label.ts (hunk 1 above)
Comment (mkzhx) [reply modules/tinymce/src/themes/silver/main/ts/ui/dialog/Label.ts]: After adding center alignment, it became cleaner to not inline the conditional classes.

**Review by ArvinJ-H — APPROVED** [review]

**Pull request merged**